app/views/matches/show.html.erb

```diff
@@ -0,0 +1,63 @@
+<div>
+  <h4><%= @my_item.name %> &amp; <%= @other_item.name %></h4>
+<!--   <%= render @other_item %> -->
+
+  <%= form_for @my_item, method: "PATCH" do |f| %>
+    <%= f.hidden_field :swapped %>
+    <%= hidden_field_tag "match", "#{@match.id}" %>
```

Comment:
I don't think you need the interpolation here - if you need a string because an int is misinterpreted then use match.id.to_s to make that explicit.